ProcessLib/HeatTransport/HeatTransportProcess.cpp

```diff
@@ -0,0 +1,94 @@
+
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "HeatTransportProcess.h"
+
+#include <cassert>
+
+#include "ProcessLib/Utils/CreateLocalAssemblers.h"
+
+namespace ProcessLib
+{
+namespace HeatTransport
+{
+HeatTransportProcess::HeatTransportProcess(
+    MeshLib::Mesh& mesh,
+    Base::NonlinearSolver& nonlinear_solver,
+    std::unique_ptr<Base::TimeDiscretization>&& time_discretization,
+    std::vector<std::reference_wrapper<ProcessVariable>>&& process_variables,
+    HeatTransportProcessData&& process_data,
+    SecondaryVariableCollection&& secondary_variables,
+    ProcessOutput&& process_output)
+    : Process(mesh, nonlinear_solver, std::move(time_discretization),
+              std::move(process_variables), std::move(secondary_variables),
+              std::move(process_output)),
+      _process_data(std::move(process_data))
+{
+    if (dynamic_cast<NumLib::ForwardEuler*>(
+            &Base::getTimeDiscretization()) != nullptr)
+    {
+        OGS_FATAL(
+            "HeatTransportProcess can not be solved with the ForwardEuler"
+            " time discretization scheme. Aborting");
+        // Because the M matrix is not assembled. Thus, the linearized system
```

Comment:
Remove these comments because the mass matrix, M, is assembled in this computation.